src/plugins/smith_waterman/src/SmithWatermanAlgorithm.cpp

```diff
@@ -173,7 +173,7 @@ void SmithWatermanAlgorithm::calculateMatrixForMultipleAlignmentResult() {
 
     n = pat_n * 2;
     int dirn = (4 + pat_n + 3) >> 2;
-    int *buf, *matrix = (int *)malloc(n * sizeof(int) + pat_n * 0x80 + matrixLength * dirn);
+    int *buf, *matrix = new int[n * sizeof(int) + pat_n * 0x80 + matrixLength * dirn];
```

Comment:
Not like this. When we use "malloc" we need to pass the exact memory value (in bytes) we need to allocate as an argument, usually it looks like:
`malloc(size_of_array * syze_of_one_element_of_array_in_bytes)`
And when we use "new" we pass just size of the array,  like:
`new[size_of_array]`
Let's figure our case out:
`malloc(n * sizeof(int) + pat_n * 0x80 + matrixLength * dirn)`
`n * sizeof(int)` - **n** is size of araray, **sizeof(int)** - sizeof is a function, which calculates size of the object (or of the type, _int_, in our case)
`pat_n * 0x80 + matrixLength * dirn` - some additional information which is stored in this array (I hate the guy who did this, he should use separate variables)
So, we need to decrease the array size, I think like this:
`new int[n + ((pat_n * 0x80 + matrixLength * dirn) / sizeof(int))];`

PS: test it!

Reply:
new qint8[original size]  looks simpler ?